# Lambda School Data Science Module 124

## Introduction to Bayesian Inference




### 1) In your own words, summarize the difference between Bayesian and Frequentist statistics.

Frequentist statistics looks purely at data to make conclusions, while Bayesian statistics uses data with current beliefs about about that data to inform the calculation.

###For questions 2) - 8) use the following information:

Let A be the event of surviving the sinking of the Titanic and B be the event of being a 1st class passenger.

In [1]:
import pandas as pd
import numpy as np


data_url = 'https://raw.githubusercontent.com/Chelsea-Myers/Lambda-Intro/master/Titanic.csv'

df = pd.read_csv(data_url, skipinitialspace=True, header=0)

print(df.shape)
df.head()

(887, 8)


Survived  Pclass  ... Parents/Children_Aboard     Fare
0         0       3  ...                       0   7.2500
1         1       1  ...                       0  71.2833
2         1       3  ...                       0   7.9250
3         1       1  ...                       0  53.1000
4         0       3  ...                       0   8.0500

[5 rows x 8 columns]

2) Calculate P(A) = P(Surving)

In [24]:
### YOUR CODE HERE
surviving = df['Survived'].value_counts(ascending=False)[1] / df['Survived'].value_counts().sum()
surviving

0.3855693348365276

3) Calculate P(B) = P(Being a 1st class passenger)

In [34]:
### YOUR CODE HERE
first_class = df['Pclass'].value_counts()[1] / df['Pclass'].count()
first_class

0.24351747463359638

4) Calculate P(A $\cap$ B) = P(Being a 1st class passenger and surviving)

In [37]:
### YOUR CODE HERE
survived_and_fc = df[df['Survived'] == 1]['Pclass'].value_counts()[1] / df['Survived'].count()
survived_and_fc

0.15332581736189402

5) Calculate P(A|B) = P(Surviving given that the person was a 1st class passenger)

In [38]:
### YOUR CODE HERE
survived_given_fc = survived_and_fc / first_class
survived_given_fc

0.6296296296296297

6) Calculate (B|A) = P(Being a first class passenger given the person survived)

In [43]:
### YOUR CODE HERE
fc_given_survived = survived_and_fc / surviving
fc_given_survived

0.39766081871345027

7) Show that 

$$P(A|B) = \frac{P(A \cap B)}{P(B)}$$

with the numbers you calculate above.

In [44]:
### YOUR CODE HERE
survived_given_fc == survived_and_fc / first_class

True

8) Show that 

$$P(A|B) = \frac{P(B|A)P(A)}{P(B)}$$

with the numbers you calculated above.

In [45]:
### YOUR CODE HERE
survived_given_fc == (fc_given_survived * surviving) / first_class

True

###Use the following information to answer questions 9) - 11) 

Assume that the passengers on the Titanic are representative of all individuals traveling from Western Europe to the United States in the early 20th century.

Let's use the Titanic data sample to estimate the mean fare paid by all individuals traveling from Western Euorpe to the United States during that time.

9) Use a 95% frequentist CI to estimate the mean fare paid by individuals traveling from Western Europe to the US in the early 20th century.

Interpret your confidence interval.

In [47]:
### YOUR CODE HERE
from scipy.stats import t

fare_mean = df['Fare'].mean()
fare_std = df['Fare'].std()
fare_n = df['Fare'].count()
fare_se = fare_std / fare_n**(1/2)

t.interval(alpha= 0.95, df=fare_n - 1, loc=fare_mean, scale=fare_se)

(29.02482490325031, 35.586015457516254)

We are 95% confident that the population mean of the fare paid by individuals traveling from Western Europe to the US in the early 20th century is between \$29.02 and \$35.59.

10) Use a 95% Bayesian CI to determine the distribution of the fare paid by individuals traveling from Western Europe to the US in the early 20th century.  Interpret your results

In [48]:
### YOUR CODE HERE
from scipy import stats

stats.bayes_mvs(df['Fare'], alpha=0.95)

(Mean(statistic=32.30542018038331, minmax=(29.024824903250337, 35.58601545751628)),
 Variance(statistic=2483.858450355689, minmax=(2262.747559034253, 2726.2430731483255)),
 Std_dev(statistic=49.824230556814946, minmax=(47.568346187714496, 52.213437668365835)))

We are 95% confident that the true mean fare paid by individuals from Western Europe to the US in the early 20th century can be described by a statistical distribution that has a mean of \$32.31 and a standard deviation of \$49.82.

#Portfolio Project Assignment

Watch the Porfolio Project websites mini-lecture: https://www.youtube.com/watch?v=gxbNCGERUjc&feature=youtu.be&ab_channel=LambdaSchool


Begin to set up your own Portfolio Project website.